src/libphonenumber/PhoneNumber.php

```diff
@@ -546,13 +546,17 @@ public function unserialize($serialized)
     {
         $data = unserialize($serialized);
 
-        $this->setCountryCode($data[0]);
-        $this->setNationalNumber($data[1]);
-        $this->setExtension($data[2]);
-        $this->setItalianLeadingZero($data[3]);
-        $this->setNumberOfLeadingZeros($data[4]);
-        $this->setRawInput($data[5]);
-        $this->setCountryCodeSource($data[6]);
-        $this->setPreferredDomesticCarrierCode($data[7]);
+        list (
+            $this->countryCode,
+            $this->nationalNumber,
+            $this->extension,
+            $this->italianLeadingZero,
+            $this->numberOfLeadingZeros,
+            $this->rawInput,
+            $this->countryCodeSource,
+            $this->preferredDomesticCarrierCode
+        ) = $data;
+
+        $this->hasNumberOfLeadingZeros = true;
```

Comment:
Just to clarify, in its current state, this PR currently does not break anything, as unserialize() always sets `$hasNumberOfLeadingZeros` to true by calling `setNumberOfLeadingZeros()`.

What I'm confused with is that the current code does not properly restore the state of `$hasNumberOfLeadingZeros`:

    $n = new libphonenumber\PhoneNumber();
    var_export($n); // 'hasNumberOfLeadingZeros' => false
    var_export(unserialize(serialize($n))); // 'hasNumberOfLeadingZeros' => true